boards/DIM/Src/App/states/App_DriveState.c

```diff
@@ -2,6 +2,63 @@
 
 #include "App_SharedMacros.h"
 #include "App_SevenSegDisplays.h"
+#include "App_ErrorCode.h"
+
+static void App_SetPeriodicCanSignals_DriveMode(
+    struct DimCanTxInterface *can_tx,
+    uint32_t                  switch_position)
+{
+    switch (switch_position)
+    {
+        case 0:
+        {
+            App_CanTx_SetPeriodicSignal_DRIVE_MODE(
+                can_tx,
+                CANMSGS_DIM_ROTARY_SWITCH_DRIVE_MODE_DRIVE_MODE_1_CHOICE);
+        }
+        break;
+        case 1:
+        {
+            App_CanTx_SetPeriodicSignal_DRIVE_MODE(
+                can_tx,
+                CANMSGS_DIM_ROTARY_SWITCH_DRIVE_MODE_DRIVE_MODE_2_CHOICE);
+        }
+        break;
+        case 2:
+        {
+            App_CanTx_SetPeriodicSignal_DRIVE_MODE(
+                can_tx,
+                CANMSGS_DIM_ROTARY_SWITCH_DRIVE_MODE_DRIVE_MODE_3_CHOICE);
+        }
+        break;
+        case 3:
+        {
+            App_CanTx_SetPeriodicSignal_DRIVE_MODE(
+                can_tx,
+                CANMSGS_DIM_ROTARY_SWITCH_DRIVE_MODE_DRIVE_MODE_4_CHOICE);
+        }
+        break;
+        case 4:
+        {
+            App_CanTx_SetPeriodicSignal_DRIVE_MODE(
+                can_tx,
+                CANMSGS_DIM_ROTARY_SWITCH_DRIVE_MODE_DRIVE_MODE_5_CHOICE);
+        }
+        break;
+        case 5:
+        {
+            App_CanTx_SetPeriodicSignal_DRIVE_MODE(
+                can_tx,
+                CANMSGS_DIM_ROTARY_SWITCH_DRIVE_MODE_DRIVE_MODE_INVALID_CHOICE);
+        }
+        break;
+        default:
+        {
+            // Should never reach here
```

Comment:
should we set it to `CANMSGS_DIM_ROTARY_SWITCH_DRIVE_MODE_DRIVE_MODE_INVALID_CHOICE` here as well?

Reply:
Hmm I think if we were to do that I'd add another enum. `case 5` is still a valid switch position that we can physically switch to (It's "invalid" only because `specs.md` says we only need 5 drive modes, not 6), but `case 6` and above would not be physically possible is likely a result of a programming or hardware error.

Reply:
OK, I'm find with it as-is then.

Reply:
👍 I'll also leave in a comment about `case 5` to further clarify what `INVALID` means